packages/compiler/src/source.ts

```diff
@@ -25,15 +26,96 @@ export async function fetchComponentSources(
     };
   };
 
-  const keys = componentPaths.map(
-    (p) => p.split('/').join(`/${SOCIAL_COMPONENT_NAMESPACE}/`) + '/*'
-  );
+  let aggregatedResponses;
+
+  if (enableBlockHeightVersioning) {
+    /**
+     * Requested components mapped by the block heights to reduce the amount of social requests
+     * If no block height specified - the "" key is used
+     */
+    const componentsByBlockHeight = componentPaths.reduce(
+      (pathsByBlockHeight, componentPath) => {
+        const [path, blockHeight] = componentPath.split('@');
+        const blockHeightKey = blockHeight || '';
+
+        if (!pathsByBlockHeight[blockHeightKey]) {
+          pathsByBlockHeight[blockHeightKey] = [];
+        }
+
+        pathsByBlockHeight[blockHeightKey].push(
+          path.split('/').join(`/${SOCIAL_COMPONENT_NAMESPACE}/`) + '/*'
+        );
+
+        return pathsByBlockHeight;
+      },
+      {} as Record<string, string[]>
+    );
+
+    const componentsByBlockHeightArr = Object.entries(componentsByBlockHeight);
+
+    const responsesWithBlockHeight = await Promise.all(
+      componentsByBlockHeightArr.map(async ([blockId, keys]) => {
```

Comment:
```suggestion
    const responsesWithBlockHeight = await Promise.all(
      Object.entries(componentsByBlockHeight).map(async ([blockId, keys]) => {
```

Reply:
yeah, good catch!
It made sense to use `componentsByBlockHeightArr` in the previous implementation, when the block height was mapped by the `idx` (array item number). Now it is just a redundant variable declaration.